### ISSUE TITLE
Error AADSTS500011 - Unsure how to fix

### ISSUE THREAD

**Issue by shiftur** [microsoft/fhir-server#3275]
**TLDR : I tried to configure this server for Smart on FHIR apps. In Postman, the Oauth2 v2 routes worked, and so did the v1 routes. But the library I am using follows the v2 routes' structure but this FHIR Server seems to only work with the v1 routes, which makes the auth fail when trying it on the Smart on FHIR app with the error code AADSTS500011. Is it because of a bad configuration on my part?**

Hi, I have been trying to setup this FHIR Server on a Azure trial account and have been unable to configure it properly. 

I have a smart on fhir app running on localhost:4200 that I'm trying to auth using the fhirclient javascript library. 

On the Azure side, I have the Web App of the FHIR Server and 2 app registrations in Azure Active Directory.
I feel like this issue is linked to my auth configs, so here they are : 

- Web App Configuration : 

```
{
    "name": "FhirServer__Security__Authentication__Audience",
    "value": "api://[Resource App Registration's Application (client) ID]",
    "slotSetting": false
  },
  {
    "name": "FhirServer__Security__Authentication__Authority",
    "value": "https://login.microsoftonline.com/[TenantID]",
    "slotSetting": false
  },
  {
    "name": "FhirServer__Security__EnableAadSmartOnFhirProxy",
    "value": "True",
    "slotSetting": false
  },
  {
    "name": "FhirServer__Security__Enabled",
    "value": "True",
    "slotSetting": false
  }
```

- Azure Active Directory : 
I added all the roles found in [https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Shared.Web/roles.json](https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Shared.Web/roles.json), added them to my user, added them to my client app registration and my resource app registration and granted both Admin consent.

I added a scope to both applications. I added the proper redirect URIs to the client app registration, including the /AadSmartOnFhirProxy/callback/[tenantID]. 

I may have left out some steps, but I mostly followed the many documents and issues found in this github.

As stated in the TLDR earlier, I made this work using Postman with both the v1 and v2 routes (v1 requiring a Ressource parameter, v2 using scopes), but the javascript library I use doesn't allow me to set a Resource parameter. 

Here is the javascript code I am currently using in the client : 

```
FHIR.oauth2.authorize({
      "client_id": "[Aad Client App registration's Application (client) ID]",
      "scope": "api://[Aad Client App registration's Application (client) ID]/.default",
      "iss": "https://[name of my FHIR server].azurewebsites.net",
      "redirectUri" : "http://localhost:4200/patientList"
    })
```

I hope this was understandable, English is not my first language.
Thanks in advance!

AB#103140

**Comment by shiftur** [comment]
I just realized one of the requests during the auth process automatically adds the resource parameter. So that probably is not the source of the error.

But in Postman, the resource that worked I was using was "api://[Aad Resource App registration's Application (client) ID]
And the one being added automatically is "https://[name of my FHIR server].azurewebsites.net".

**Comment by shiftur** [comment]
Fixed by changing the domain of the app registration and the FHIR server itself
